kolibri/plugins/management/assets/src/device_management/deviceApp.js

```diff
@@ -0,0 +1,71 @@
+import KolibriModule from 'kolibri_module';
+import { getCurrentSession } from 'kolibri.coreVue.vuex.actions';
+import router from 'kolibri.coreVue.router';
+import Vue from 'kolibri.lib.vue';
+import RootVue from './views';
+import store from './state/store';
+import { PageNames } from './constants';
+import preparePage from '../state/preparePage';
+import {
+  showManagePermissionsPage,
+  showUserPermissionsPage,
+} from './state/actions/managePermissionsActions';
+import { showManageContentPage } from './state/actions/manageContentActions';
+
+function hideLoadingScreen() {
+  store.dispatch('CORE_SET_PAGE_LOADING', false);
+}
+
+const routes = [
+  {
+    path: '/',
+    redirect: '/content',
+  },
+  {
+    name: PageNames.MANAGE_CONTENT_PAGE,
+    path: '/content',
+    handler: ({ name }) => {
+      preparePage(store.dispatch, {
+        name,
+        title: 'Manage Device Content',
+      });
+      showManageContentPage(store).then(hideLoadingScreen);
+    },
+  },
+  {
+    name: PageNames.MANAGE_PERMISSIONS_PAGE,
+    path: '/permissions',
+    handler: ({ name }) => {
+      preparePage(store.dispatch, {
+        name,
+        title: 'Manage Device Permissions',
+      });
+      showManagePermissionsPage(store).then(hideLoadingScreen);
+    },
+  },
+  {
+    name: PageNames.USER_PERMISSIONS_PAGE,
+    path: '/permissions/:userid',
+    handler: ({ params, name }) => {
+      preparePage(store.dispatch, {
+        name,
+        title: 'Manage User Permissions',
+      });
+      showUserPermissionsPage(store, params.userid).then(hideLoadingScreen);
+    },
+  },
+];
+
+class DeviceManagementModule extends KolibriModule {
+  ready() {
+    getCurrentSession(store).then(() => {
+      this.rootvue = new Vue({
+        el: 'rootvue',
+        render: createElement => createElement(RootVue),
+        router: router.init(routes),
+      });
+    });
+  }
+}
+
+export default new DeviceManagementModule();
```

Comment:
The location and name of the `device_management` app is confusing to me. Typically all other top-level URLs and apps are an independent top-level Kolibri plugin (`/learn`, `/user`, etc). This one is nested under `management`.

Longer-term, I'd love to revisit this pattern because it adds a lot of boilerplate. However I think it would be best to maintain consistency for now and refactor holistically in the future.

Reply:
@jonboiser and I had discussed this before he began implementing, and it made sense to me to do it in this way, as it was essentially a splitting of the management app into a facility management and device management app.

The only limitation of lumping apps together in the same plugin folder like this is a limitation in enabling/disabling.